### ISSUE TITLE
About Appropriate migration time to move Caffe2 from Caffe

### ISSUE THREAD

**Issue by leemgs** [facebookarchive/caffe2#862]
Dear All, 

There are many existing projects based on the existing version, Caffe. 
After Facebook open sourced Caffe2 (Apr 18, 2017) as a next version of Caffe, It seems that
Caffe2 include  lots of new features (https://caffe2.ai/docs/caffe-migration.html#how-does-caffe-compare-to-caffe2) as following: 

* Caffe2 improves Caffe 1.0 in a series of directions:
* first-class support for large-scale distributed training
* mobile deployment
* new hardware support (in addition to CPU and CUDA)
* flexibility for future directions such as quantized computation
* stress tested by the vast scale of Facebook applications

The official website provides how to convert from Caffe as https://caffe2.ai/docs/caffe-migration.html#caffe-to-caffe2. Nvidia also has already started Caffe2 related instruction as a 3rd party at https://developer.nvidia.com/caffe2. 

At this time, does it reasonable time that we try to change a base deep learning framework from Caffe to Caffe2 in case of a **commercial** goal? Where can we find a detail comparison of two versions that is compared between Caffe and Caffe2? 

* Caffe: https://github.com/BVLC/caffe ,BSD license,  4,034 commits , 247 contributors, initial release (v0.1): 19-Mar-2014, 14 releases
* Caffe2: https://github.com/caffe2/caffe2, BSD license, 1,854 commits , 88 contributors, initial release (v0.6.0): 4-Apr-2017, 2 releases

**Comment by bgchun** [comment]
IMO, if you plan to use DNNs for production systems, in particular mobile, definitely caffe2.

**Comment by leemgs** [comment]
@bgchun , Thank you for your opinion.

**Comment by leemgs** [comment]
To Caffe2-based developers and commercial users

I think that Facebook and lots of contributors have submitted their patches to deploy Caffe2 as a commercial deep learning framework (= production edge deployment) against Caffe. In case of Android mobile device, we could read an official instruction as following:

* https://caffe2.ai/docs/mobile-integration.html
* https://github.com/caffe2/caffe2/blob/master/scripts/build_android.sh
* https://caffe2.ai/docs/AI-Camera-demo-android.html (AI Camera Demo and Tutorial)

Anyone who already tried to migrate a base deep learning framework from Caffe to a Caffe2-based project?

**Comment by leemgs** [comment]
\CC: @myungjoo , @jijoongmoon , @wonsubkim

**Comment by myungjoo** [comment]
Yes, as we've guessed, we should move on to Caffe2 as soon as possible.

Let's look at how we will be able to implement "Caffe2 Driver" for specific hardware and how we can access Caffe2 from our API sets.

**Comment by leemgs** [comment]
> https://github.com/caffe2/caffe2/issues/274#issuecomment-295381560

@Yangqing, @akyrola, Please, have a look at this. Could you give me your experience or opinion about this topic?
